modelator/cli/__init__.py

```diff
@@ -0,0 +1,232 @@
+
+    if model_path is None:
+        # reload current model
+        model = ModelFile.load(LOG_LEVEL)
+        if model is None:
+            print("Model file does not exist")
+            return
+        
+        model_path = model.tla_file_path
+
+    print(f"Loading {model_path}... ")
+    model = _create_and_parse_model(model_path)
+    ModelFile.save(model)
+    print('Loading OK ✅')
+
+
+@app.command()
+def typecheck():
+    '''
+    Type check the loaded model, if available.
+    '''
+    global LOG_LEVEL
+    model = ModelFile.load(LOG_LEVEL)
+    if model is None:
+        print("Model file does not exist")
+        return
+    
+    try:
+        model.typecheck()
+        print('Type checking OK ✅')
+    except ModelError as e:
+        print('Type checking error 💥')
+        print(e)
+
+
+@app.command()
+def check(
+    config_path: Optional[str] = typer.Option(None, help="Path to TOML file with the model and model checker configuration."), 
+    model_path: Optional[str] = typer.Option(None, help="Path to the TLA+ model file (overwrites config file)."),
+    constants: Optional[List[str]] = typer.Option(None, help="Constant definitions in the format 'name=value' (overwrites config file)."), 
+    invariants: Optional[List[str]] = typer.Option(None, help="List of invariants to check (overwrites config file)."),  
+):
+    '''
+    Check that the invariants hold in the model, or generate a trace for a counterexample.
+    '''
+    mc_invariants = invariants
+    if config_path is None and mc_invariants == []:
+        print("Error: either --config-path or --invariants must be specified.")
+        raise typer.Exit(code=1)
+
+    # Dict is not supported by typer
+    constants = dict([c.split("=") for c in constants])
+
+    config = load_config_file(config_path)
+    model_path = config['model_path'] if model_path is None else model_path
+    constants = config['constants'] if constants is None else constants
+    mc_invariants = config['invariants'] if mc_invariants is None else mc_invariants
+    init = config['init']
+    next = config['next']
+
+    model = _load_model(model_path, init, next, constants)
+    if model is None:
+        return    
+    
+    start_time = timer()
+    result = model.check(mc_invariants, constants)
+    _print_results(result)
+    print(f"Total time: {(timer() - start_time):.2f} seconds")
+
+
+@app.command()
+def sample(
+    config_path: Optional[str] = typer.Option(None, help="Path to TOML file with the model and model checker configuration."), 
+    model_path: Optional[str] = typer.Option(None, help="Path to the TLA+ model file (overwrites config file)."),
+    constants: Optional[List[str]] = typer.Option(None, help="Constant definitions in the format 'key=value' (overwrites config file)."), 
+    desired_states: Optional[List[str]] = typer.Option(None, help="Model operators describing desired states in the model execution (overwrites config file)."), 
```

Comment:
I don't think `desired_states` is the right name. The operator can describe the shape of the trace using [state, action, trace invariants](https://apalache.informal.systems/docs/apalache/principles/invariants.html). 

I would simply call it `examples`, on par with `invariants` for the `check` command.